UI/win-update/updater/updater.cpp

```diff
@@ -1546,6 +1579,10 @@ static bool Update(wchar_t *cmdLine)
 			DeleteFile(update.tempPath.c_str());
 	}
 
+	/* Delete all removed files mentioned in the manifest */
+	for (wstring file : deletions)
+		DeleteFile(file.c_str());
```

Comment:
Should be simple enough, although I do wonder how that flag behaves nowadays where people rarely actually reboot Windows and it instead goes into this "deep sleep" state when you click "Shut down".

Reply:
I added another commit to add another function that mimics the behaviour of trying to rename + scheduling deletion: https://github.com/obsproject/obs-studio/pull/6500/commits/ad6f8186ea4e7350147c261983a2857140efb5e6

Not sure if that's necessarily the best way to do it since it may leave a file in place if it can't be renamed for whatever reason. Maybe this should show a warning of some kind if it fails?

Reply:
In the current case the deletion only happens after all update operations completed, so it should never delete anything prematurely. But I see the point of the random hash now.

I could perhaps add a new struct (e.g. `deletion_t`) to keep track of renames before deletions, and then extend `CleanupPartialUpdates` to roll-back renames in case one of the `MoveFile` calls fails.

Reply:
Alright updated it now to rename a file during the update process where rolling back is still possible, and then delete at the end. Also just copied the random string part from `MoveInUseFileAway()`.

(I still need to test this to make sure I didn't break anything)

Reply:
Checked the latest commit now and made sure it still works as intended.